src/grid.component.ts

```diff
@@ -221,7 +222,14 @@ export class GridComponent implements OnInit, AfterContentInit, AfterViewInit {
       this.bodyOffsetHeight = this.bodyRef.nativeElement.offsetHeight;
       this.headerTopLimit = this.bodyOffsetHeight + this.bodyOffsetTop
         - this.headerOffsetTop - this.headerOffsetHeight;
-      this.headerTop = document.body.scrollTop - this.headerOffsetTop;
+      this.parentOffset = this.headerRef.nativeElement.offsetParent.offsetTop;
+      this.headerTop = document.body.scrollTop - this.headerOffsetTop - this.parentOffset;
+
+      const banner: any = document.body.querySelector('[role="banner"]');
```

Comment:
I've used "banner" selector for avoiding cases with problems, when the floating headers are under the site headers, while scrolling.
Because, If the page have a banner, it will take away some pixels from the body height, and the floating headers will start drawing under the banner layer, so banner will be above the floating headers, and in some cases it won't be readable, while scrolling. We can avoid it, increasing z-index for the floating headers, but if we did it, floating headers would be above the banner layer, and the banner would't be readable instead. 
According to the html5 attributes, role "banner" is very common. It includes the main headline or a page title. For example, a logo and site name. Usually, it is recommended to use no more than once per page. So, if the page have a banner on the top of the page, in any situation, we won't have any problems with the floating headers offsets, because the height of the banner will be already counted. If the page don't have a banner, this case would be ignored.